lib/Doctrine/ORM/Id/AbstractIdGenerator.php

```diff
@@ -19,18 +19,18 @@
 
 namespace Doctrine\ORM\Id;
 
-use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
 
 abstract class AbstractIdGenerator
 {
     /**
      * Generates an identifier for an entity.
      *
-     * @param EntityManager $em
+     * @param EntityManagerInterface $em
      * @param object|null $entity
      * @return mixed
      */
-    abstract public function generate(EntityManager $em, $entity);
```

Comment:
This is a BC-break, that's why we didn't do it on v2.x. v3.x (`dev-master`) already solved this AFAIK.

Reply:
Ok